src/liboslexec/llvm_util.cpp

```diff
@@ -294,15 +294,17 @@ LLVM_Util::LLVM_Util (int debuglevel, int vector_width)
     m_llvm_type_matrix_ptr = (llvm::PointerType *) llvm::PointerType::get (m_llvm_type_matrix, 0);
 
     // Setup up wide aliases
-    // TODO:  why are there casts to the base class llvm::Type *?  
+    // TODO:  why are there casts to the base class llvm::Type *?
+    if (m_vector_width != 4 && m_vector_width != 8 && m_vector_width != 16)
+        m_vector_width = 4;
```

Comment:
How would user know they passed illegal/unsupported vector width, maybe report an error if an assert is not desired.

Reply:
switching to floor2(clamp(m_vector_width, 4, 16)), which effectively chooses the largest valid vector width less than or equal to what they requested.